spharpy/spherical.py

```diff
@@ -1,6 +1,431 @@
 import numpy as np
 import scipy.special as special
 import spharpy.special as _special
+import logging as logger
+from pyfar import Coordinates
+from functools import lru_cache
+from spharpy.samplings.helpers import calculate_sampling_weights
+
+
+class SphHarm:
+    def __init__(self, n_max, coords, basis_type='complex', channel_convention='acn', inv_transform_type=None,
+                 normalization='n3d'):
+        r"""
+        Create a spherical harmonics class for transformation between spherical harmonics and signals.
```

Comment:
How about
```suggestion
        Class for spherical harmonics basis functions.

        Objects of this class can be used to obtain spherical harminics basis functions, their inverse, and ...
```

Reply:
Would it make sense to mention already here, the dimension of the SH matrix?

Reply:
How about giving the exact definitions of all Spherical Harmonics Conventions that can be realized with the class at the beginning of the docstring?